fixes.inc

```diff
@@ -14281,6 +14354,133 @@ native BAD_DestroyActor(actorid) = DestroyActor;
 	#define DestroyActor__( FIXES_DestroyActor(
 #endif
 
+/**
+ * <fixes> Various bugs in GetVehicleComponentInSlot function.</fixes>
+ */
+
+#if _FIXES_SAMP && defined _ALS_GetVehicleComponentInSlot
+	#error _ALS_GetVehicleComponentInSlot defined
+#endif
+
+native BAD_GetVehicleComponentInSlot(vehicleid, slot) = GetVehicleComponentInSlot;
+
+#if FIX_GetVehicleComponentInSlot
+
+	stock FIXES_GetVehicleComponentType(componentid)
+	{
+		if(IsComponentFrontBullbar(componentid))
+		{
+			return CARMODTYPE_FRONT_BULLBAR;
+		}
+		else if(IsComponentRearBullbar(componentid))
+		{
+			return CARMODTYPE_REAR_BULLBAR;
+		}
+		return GetVehicleComponentType__(componentid);
+	}
+
+	stock FIXES_AddVehicleComponent(vehicleid, componentid)
+	{
+		switch(GetVehicleComponentType(componentid))
+		{
+			case CARMODTYPE_STEREO:
+			{
+				FIXES_gsStereo[vehicleid] = componentid;
+			}
+			case CARMODTYPE_FRONT_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][0] = componentid;
+			}
+			case CARMODTYPE_REAR_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][1] = componentid;
+			}
+			case CARMODTYPE_FRONT_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][0] = componentid;
+			}
+			case CARMODTYPE_REAR_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][1] = componentid;
+			}
+		}
+		return AddVehicleComponent__(vehicleid, componentid);
+	}
+
+	stock FIXES_RemoveVehicleComponent(vehicleid, componentid)
+	{
+		switch(GetVehicleComponentType(componentid))
+		{
+			case CARMODTYPE_STEREO:
+			{
+				FIXES_gsStereo[vehicleid] = 0;
+			}
+			case CARMODTYPE_FRONT_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][0] = 0;
+			}
+			case CARMODTYPE_REAR_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][1] = 0;
+			}
+			case CARMODTYPE_FRONT_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][0] = 0;
+			}
+			case CARMODTYPE_REAR_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][1] = 0;
+			}
+		}
+		return RemoveVehicleComponent__(vehicleid, componentid);
+	}
+
+	stock FIXES_GetVehicleComponentInSlot(vehicleid, slot)
+	{
+		switch(slot)
+		{
+			case CARMODTYPE_STEREO:
+			{
+				return FIXES_gsStereo[vehicleid];
```

Comment:
As well as the other comments about whether five variables are needed or not (which hinges on whether the code in `OnVehicleMod` is correct), this won't currently work with multiple scripts.  There are quite a few examples in the code of optionally distributed fixes.

Reply:
I don't even understand this. Why this won't work? Due to code within `OnVehicleMod` hinging?

Reply:
It won't work because if you add a component in a filterscript, getting it in a gamemode will return 0.

Reply:
So, any workaround, how can I use a variable in multiple scripts, except those I know like PVars.

Reply:
or returning it via callback (CallLocalFunction/Remote)

Reply:
Properties or `CallRemoteFunction`.  As I said, there are dozens of examples in fixes.inc.

Reply:
Done in https://github.com/Mergevos/sa-mp-fixes/commit/24299665dc6d6e88a194b2874d5c2c3c7c605210